File: v1/Demo/source/demo_get_firmware_version.cpp
#include "../include/Handapi.h"

using namespace HandProtocol;

int main()
{
    Handapi api;
    api.init(LEFT_IP);
    std::string version = api.get_firmware_version();

    if (version == "")
    {
        Logger::get_instance()->print_trace_error("get version failed\n");
        return ResultCode::FAIL;
    }

    Logger::get_instance()->print_trace("get version success: %s\n", version.c_str());

    
    return 0;
}

File: v1/Demo/demo_get_firmware_version.py
import sys, os
sys.path.append(os.path.join(os.path.dirname(__file__), "../"))
from Hand_Sdk.HandSocket import *
from Hand_Sdk.Handapi import *

def main():
    ip_left = "192.168.137.39" # lefthand
    # ip_right = "192.168.137.19" # righthand
    api = API(ip_left)
    version = api.get_firmware_version()

    if version == "":
        logger.print_trace_error("get version failed")
    else:
        logger.print_trace("get version success", version)

if __name__ == "__main__":
    main()



